changelog/fix_rails_delegate_to_handle_self_class_constants_instance_variables_and_class_variables.md

```diff
@@ -0,0 +1 @@
+* [#912](https://github.com/rubocop/rubocop-rails/issues/912): Enhance `Rails/Delegate` by adding delegation detection for `self.class`, constants, instance variables, and class variables. ([@ydakuka][])
```

Comment:
Can you add a mention of global variables?

Reply:
Done.

P.S. I got a failed test that is not related to this PR.

Reply:
The issue was fixed in https://github.com/rubocop/rubocop-rails/commit/f9adec0fc86e386504c7508105d485f3e84f9ea7. There's no problem with this PR. Thank you!